internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -256,14 +257,38 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 	return r.updateStatus(ctx, log, clusterInstall, clusterDeployment, cluster, nil)
 }
 
-func (r *ClusterDeploymentsReconciler) validateClusterDeployment(clusterDeployment *hivev1.ClusterDeployment,
-	clusterInstall *hiveext.AgentClusterInstall) error {
+// validateClusterDeployment ensures that the cluster deployment's pull secret and release image are valid
+func (r *ClusterDeploymentsReconciler) validateClusterDeployment(ctx context.Context, key types.NamespacedName, clusterDeployment *hivev1.ClusterDeployment,
+	clusterInstall *hiveext.AgentClusterInstall) (string, *models.ReleaseImage, error) {
```

Comment:
I think maybe split this into two functions `getPullSecret` and `getReleaseImage`.

It's a bit strange for a validation function to be returning these things and I can imagine it growing as more things are validated and returned.

Reply:
Great idea!! Thank you!! Modified as suggested